RELEASING.md

```diff
@@ -4,7 +4,8 @@ Release Guide
 1. Ensure that pulpcore of the corresponding version is available on PyPI.
 1. Run the release script (`python .ci/scripts/release.py`).
 1. Check the updated version in the `galaxy.yml` and in the `roles/pulp_common/{vars,defaults}/main.yml`.
-1. Check references to current and previous versions in `docs/index.md`\
+1. Check references to current and previous versions in `docs/index.md` & `docs/customizing.md`
+   (customizing.md added during 3.13 development cycle.)\
```

Comment:
`docs/index` was already bad to automate during release, adding another file makes it worse, does it really need to have always the latest version? If so, the release script needs to be updated: https://github.com/pulp/pulp_installer/blob/master/.ci/scripts/release.py

Reply:
If you say we should use words like "latest version", "stable minor version" instead of "3.12.1" or "3.12", i agree.

Reply:
@mdellweg I agree, I changed it to that.

Reply:
do we still need this line? I ctrl+F 3.12.2 and I didn't find anything, so I think release doesn't require this step anymore